src/Raven.Server/Documents/Queries/Results/TimeSeriesRetriever.cs

```diff
@@ -598,15 +603,30 @@ object GetValue(QueryExpression expression, SingleResult singleResult)
 
                             if (index >= singleResult.Values.Length)
                                 return null;
+                            if (_currentIsRaw)
+                                return singleResult.Values.Span[index];
+
+                            // we are working with a rolled-up series
+                            // here an entry has 6 different values (min, max, first, last, sum, count) per each 'original' measurement 
+                            // we need to return the average value (sum / count)
+                            index *= 6;
+                            if (index + (int)AggregationType.Count >= singleResult.Values.Length)
+                                return null;
+                            return singleResult.Values.Span[index + (int)AggregationType.Sum] / singleResult.Values.Span[index + (int)AggregationType.Count];
```

Comment:
Can this cause divide by zero?

Reply:
I don't think so.. but I added a check for this, just to be safe